src/swapd/runtime.rs

```diff
@@ -105,7 +105,9 @@ pub fn run(config: ServiceConfig, opts: Opts) -> Result<(), Error> {
         swap_id,
         tasks,
         monero_height: 0,
+        monero_checkpoint_height: None,
```

Comment:
Is this for all intents and purposes the same as the `monero_address_creation_height`? If so, I'd refactor those two concepts together.

Reply:
Looks to me like we are not actually using `monero_checkpoint_height` in practice, but instead only the `monero_address_creation_height`. 

In terms of performance, the gains are marginal at best. Re-scanning a few blocks more in Monero's case hardly makes a difference and Bitcoin does not really care about height either way. We don't expect more than one transaction anyway from each of these addresses, so that should not be a concern here.

The checkpointed height for Bitcoin and Monero is also overwritten on each checkpoint made. If we re-watch any address, it won't return a previous transaction if it happened before the last checkpoint. While it's not a problem right now, since we don't need these transactions anymore, this feels a bit brittle.

Here's what I'd propose: We add a `HashMap<TxLabel, BlockHeight> for the address creation heights and insert whenever we create an address. Then we checkpoint and restore this data structure. What do you think?